src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,61 @@
+package core.basesyntax;
+
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final String DEFAULT_COLOR = "WHITE";
+    private static final int DEFAULT_RADIUS = 10;
+    private static final int QUANITY_OF_TYPES = 5;
+    private static final int CONSTANT = 100;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private String color = colorSupplier.getRandomColor();
+    private Random random = new Random();
+    private int figureType = random.nextInt(QUANITY_OF_TYPES);
+    private double randomDouble = (double) Math.round(random.nextDouble() * CONSTANT) / CONSTANT;
```

Comment:
With current approach where randomDouble is a class variable, it never changes its value.
So all your "random" figures will have absolutely same dimentions
Please make
private double getRandomDouble() method and move this logic there
So each time method is called it will provide new value